pkg/logs/decoder/matcher.go

```diff
@@ -0,0 +1,22 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2019 Datadog, Inc.
+
+package decoder
+
+// EndLineMatcher defines the criterion to whether to end a line or not.
+type EndLineMatcher interface {
+	// Match takes the existing bytes and the bytes to be appended, returns
+	// true if the combination matches the end of line condition.
+	Match(exists []byte, appender []byte, start int, end int) bool
```

Comment:
I don't mind using bytes.Buffer instead of []byte (I don't know if the `d.lineBuffer.Bytes()` does an allocation of not)
I was wondering if we need this variable at all, or if we can just use `(appender []byte, start int, end int)`, because I think the Docker tailer is going to send full line. But that's a assumption we probably don't want to make